apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,40 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
+export class CardComponent<T extends { id: number }> {
+  @Input() list: T[] | null = null;
+
   @Input() customClass = '';
 
-  CardType = CardType;
+  @Output() deleteItem = new EventEmitter<number>();
+
+  @Output() addItem = new EventEmitter();
 
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore
-  ) {}
+  @ContentChild(TemplateRef, { static: true }) templateRef: TemplateRef<{
+    item: T;
+  }> | null = null;
```

Comment:
you should name your template inside the parent, using a directive. If more template get added, you may run into troubles